code/modules/economy/Accounts_DB.dm

```diff
@@ -1,247 +1,177 @@
-	if(!ui)
-		ui = new(user, src, ui_key, "AccountsUplinkTerminal", name, 800, 600, master_ui, state)
-		ui.open()
-
-/obj/machinery/computer/account_database/ui_data(mob/user)
-	var/list/data = list()
-	data["currentPage"] = current_page
-	data["is_printing"] = (next_print > world.time)
-	ui_login_data(data, user)
-	if(data["loginState"]["logged_in"])
-		switch(current_page)
-			if(AUT_ACCLST)
-				var/list/accounts = list()
-				for(var/i in 1 to length(GLOB.all_money_accounts))
-					var/datum/money_account/D = GLOB.all_money_accounts[i]
-					accounts.Add(list(list(
-						"account_number" = D.account_number,
-						"owner_name" = D.owner_name,
-						"suspended" = D.suspended ? "SUSPENDED" : "Active",
-						"account_index" = i)))
-
-				data["accounts"] = accounts
-
-			if(AUT_ACCINF)
-				data["account_number"] = detailed_account_view.account_number
-				data["owner_name"] = detailed_account_view.owner_name
-				data["money"] = detailed_account_view.money
-				data["suspended"] = detailed_account_view.suspended
-
-				var/list/transactions = list()
-				for(var/datum/transaction/T in detailed_account_view.transaction_log)
-					transactions.Add(list(list(
-						"date" = T.date,
-						"time" = T.time,
-						"target_name" = T.target_name,
-						"purpose" = T.purpose,
-						"amount" = T.amount,
-						"source_terminal" = T.source_terminal)))
-
-				data["transactions"] = transactions
-	return data
-
-
-/obj/machinery/computer/account_database/ui_act(action, list/params)
-	if(..())
+/datum/money_account_database/can_vv_delete()
+	message_admins("An admin attempted to VV delete a money account database, this will break the economy system for the round, if you know what you are doing please use advanced proccall")
+	return FALSE
+
+/datum/money_account_database/proc/create_account(account_name = "Unnamed", starting_funds = 0, _security_level = ACCOUNT_SECURITY_ID, terminal, supress_log = FALSE)
+	var/datum/money_account/new_account = new(account_name, starting_funds, ACCOUNT_SECURITY_ID)
+	user_accounts += new_account
+	new_account.database_holder = src
+	if(!supress_log)
+		log_account_action(new_account, starting_funds, "Account Creation", terminal, log_on_database = TRUE)
+	return new_account
+
+/datum/money_account_database/proc/create_vendor_account()
+	if(vendor_account)
+		log_debug("create_vendor_account() was called but a vendor account already exists")
 		return
-
-	. = TRUE
-
-	if(ui_login_act(action, params))
+	var/datum/money_account/new_vendor_account = new("[station_name()] Vendor Account", DEPARTMENT_BASE_PAY_MEDIUM, ACCOUNT_SECURITY_RESTRICTED)
+	vendor_account = new_vendor_account
+	vendor_account.database_holder = src
+
+///takes in an account_numb and returns either an account if it locates one or null if it finds none
+/datum/money_account_database/proc/find_user_account(account_number, include_departments = FALSE)
+	for(var/datum/money_account/account in user_accounts)
```

Comment:
Maybe a hashmap would be wise for lookups like these. Not sure what the performance impact is though

Reply:
the best way would probably just be to have a regular numbered array and directly access an index based on the account number since the last three digits will never be random and can actually be mapped to the order in which the account was created